homeassistant/components/rfxtrx/device_action.py

```diff
@@ -0,0 +1,86 @@
+"""Provides device automations for RFXCOM RFXtrx."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_TYPE
+from homeassistant.core import Context, HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+from . import DATA_RFXOBJECT, DOMAIN
+from .helpers import async_get_device_object
+
+CONF_DATA = "data"
+CONF_SUBTYPE = "subtype"
+
+ACTION_TYPE_COMMAND = "send_command"
+ACTION_TYPE_STATUS = "send_status"
+
+ACTION_TYPES = {
+    ACTION_TYPE_COMMAND,
+    ACTION_TYPE_STATUS,
+}
+
+ACTION_SELECTION = {
+    ACTION_TYPE_COMMAND: "COMMANDS",
+    ACTION_TYPE_STATUS: "STATUS",
+}
+
+ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_TYPE): vol.In(ACTION_TYPES),
+        vol.Required(CONF_SUBTYPE): str,
+        vol.Required(CONF_DATA): int,
+    }
+)
+
+
+async def async_get_actions(hass: HomeAssistant, device_id: str) -> list[dict]:
+    """List device actions for RFXCOM RFXtrx devices."""
+    actions = []
+
+    device = async_get_device_object(hass, device_id)
+    if device:
+        for action_type in ACTION_TYPES:
+            if hasattr(device, action_type):
+                values = getattr(device, ACTION_SELECTION[action_type], {})
+                for key, value in values.items():
+                    actions.append(
+                        {
+                            CONF_DEVICE_ID: device_id,
+                            CONF_DOMAIN: DOMAIN,
+                            CONF_TYPE: action_type,
+                            CONF_SUBTYPE: value,
+                            CONF_DATA: key,
+                        }
+                    )
+                actions.append(
+                    {
+                        CONF_DEVICE_ID: device_id,
+                        CONF_DOMAIN: DOMAIN,
+                        CONF_TYPE: action_type,
+                        CONF_SUBTYPE: "...",
+                    }
```

Comment:
This will show up as "Send command: ..." / "Send action: ..." in the frontend, is it intentional?

Reply:
It is intentional. To allow sending of a command that is not yet listed in the backend lib or part of a device type that don't have any listed commands.